temporalio/worker/_workflow.py

```diff
@@ -250,9 +300,32 @@ async def _handle_activation(
                         activate_task, self._deadlock_timeout_seconds
                     )
                 except asyncio.TimeoutError:
-                    raise RuntimeError(
-                        f"[TMPRL1101] Potential deadlock detected, workflow didn't yield within {self._deadlock_timeout_seconds} second(s)"
-                    )
+                    # extract the stack of the workflow if possible
+
+                    # cases:
+                    # 1. get_worker_ident() is not implemented for the worker
+                    # 2. get_worker_ident() returns None
+                    # 3. tid is not present in sys._current_frames()
+                    # 4. error occurs during traceback construction
+                    # 5. normal
+
+                    msg = f"[TMPRL1101] Potential deadlock detected, workflow didn't yield within {self._deadlock_timeout_seconds} second(s)"
+                    try:
+                        tid = workflow.get_worker_ident()
+                    except NotImplementedError:
+                        # Default error if the function is not implemented for the workers
+                        raise RuntimeError(msg) from None
```

Comment:
Yeah so here instead of having to catch NotImplemented you can just check if tid is None if you change the default implementation per my other suggestion.